packages/grafana-i18n/src/i18n.tsx

```diff
@@ -0,0 +1,159 @@
+
+    // If translations are empty strings (no translation), fall back to the default value in source code
+    returnEmptyString: false,
+
+    // Required to ensure that `resolvedLanguage` is set property when an invalid language is passed (such as through 'detect')
+    supportedLngs: LANGUAGES.map((language) => language.code),
+    fallbackLng: DEFAULT_LANGUAGE,
+
+    ns,
+    postProcess: [
+      // Add pseudo processing even if we aren't necessarily going to use it
+      PSEUDO_LOCALE,
+    ],
+  };
+
+  if (language === 'detect') {
+    getI18nInstance().use(LanguageDetector);
+    const detection: DetectorOptions = { order: ['navigator'], caches: [] };
+    options.detection = detection;
+  } else {
+    options.lng = LANGUAGES.find((lang) => lang.code === language)?.code ?? undefined;
+  }
+
+  if (module) {
+    getI18nInstance().use(module).use(initReactI18next); // passes i18n down to react-i18next
+  } else {
+    getI18nInstance().use(initReactI18next); // passes i18n down to react-i18next
+  }
+
+  if (process.env.NODE_ENV === 'development') {
+    const { default: Pseudo } = await import('i18next-pseudo');
+    getI18nInstance().use(
+      new Pseudo({
+        languageToPseudo: PSEUDO_LOCALE,
+        enabled: true,
+        wrapped: true,
+      })
+    );
+  }
+
+  await getI18nInstance().init(options);
+
+  tFunc = getI18nInstance().t;
+  transComponent = (props: TransProps) => <I18NextTrans shouldUnescape ns={ns} {...props} />;
+
+  return {
+    language: getI18nInstance().resolvedLanguage,
+  };
+}
+
+export function getLanguage() {
+  return getI18nInstance()?.language || DEFAULT_LANGUAGE;
+}
+
+export function getResolvedLanguage() {
+  return getI18nInstance()?.resolvedLanguage || DEFAULT_LANGUAGE;
+}
+
+export function getNamespaces() {
+  return getI18nInstance()?.options.ns;
+}
+
+export async function changeLanguage(language?: string) {
+  await getI18nInstance().changeLanguage(language ?? DEFAULT_LANGUAGE);
+}
+
+type ResourceKey = string;
+type ResourceLanguage = Record<string, ResourceKey>;
+type ResourceType = Record<string, ResourceLanguage>;
+
+export function addResourceBundle(language: string, namespace: string, resource: ResourceType) {
+  getI18nInstance().addResourceBundle(language, namespace, resource, undefined, true);
+}
+
+export function t(id: string, defaultMessage: string, values?: Record<string, unknown>) {
+  if (!tFunc) {
+    if (process.env.NODE_ENV !== 'test') {
+      console.warn(
+        't() was called before i18n was initialized. This is probably caused by calling t() in the root module scope, instead of lazily on render'
+      );
+    }
+
+    if (process.env.NODE_ENV === 'development') {
+      throw new Error('t() was called before i18n was initialized');
+    }
+
+    tFunc = getI18nInstance().t;
+  }
+
+  return tFunc(id, defaultMessage, values);
+}
+
+export function useTranslate() {
+  return t;
+}
+
+export const Trans = (props: TransProps): ReactElement => {
```

Comment:
could we just let typescript infer the return type here? 🤔 or if we use `ReactElement` directly, i think we have to list `react` as a `devDep`. right now it's sorta working by accident (some other package depends on `react`, so it just so happens to be there)

Reply:
Sounds good to me 👍

Reply:
Fixed 👍